libs/langchain-azure-cosmosdb/src/caches.ts

```diff
@@ -0,0 +1,172 @@
+  getCacheKey,
+  serializeGeneration,
+} from "@langchain/core/caches";
+import { Generation } from "@langchain/core/outputs";
+import { Document } from "@langchain/core/documents";
+import { EmbeddingsInterface } from "@langchain/core/embeddings";
+import { CosmosClient, CosmosClientOptions } from "@azure/cosmos";
+import { DefaultAzureCredential } from "@azure/identity";
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+import {
+  AzureCosmosDBNoSQLConfig,
+  AzureCosmosDBNoSQLVectorStore,
+} from "./azure_cosmosdb_nosql.js";
+
+const USER_AGENT_SUFFIX = "langchainjs-cdbnosql-semanticcache-javascript";
+const DEFAULT_DATABASE_NAME = "semanticCacheDB";
+const DEFAULT_CONTAINER_NAME = "semanticCacheContainer";
+
+/**
+ * Represents a Semantic Cache that uses CosmosDB NoSQL backend as the underlying
+ * storage system.
+ *
+ * @example
+ * ```typescript
+ * const embeddings = new OpenAIEmbeddings();
+ * const cache = new AzureCosmosDBNoSQLSemanticCache(embeddings, {
+ *   databaseName: DATABASE_NAME,
+ *   containerName: CONTAINER_NAME
+ * });
+ * const model = new ChatOpenAI({cache});
+ *
+ * // Invoke the model to perform an action
+ * const response = await model.invoke("Do something random!");
+ * console.log(response);
+ * ```
+ */
+export class AzureCosmosDBNoSQLSemanticCache extends BaseCache {
+  private embeddings: EmbeddingsInterface;
+
+  private config: AzureCosmosDBNoSQLConfig;
+
+  private cacheDict: { [key: string]: AzureCosmosDBNoSQLVectorStore } = {};
+
+  constructor(
+    embeddings: EmbeddingsInterface,
+    dbConfig: AzureCosmosDBNoSQLConfig
+  ) {
+    super();
+    let client: CosmosClient;
+
+    const connectionString =
+      dbConfig.connectionString ??
+      getEnvironmentVariable("AZURE_COSMOSDB_NOSQL_CONNECTION_STRING");
+
+    const endpoint =
+      dbConfig.endpoint ??
+      getEnvironmentVariable("AZURE_COSMOSDB_NOSQL_ENDPOINT");
+
+    if (!dbConfig.client && !connectionString && !endpoint) {
+      throw new Error(
+        "AzureCosmosDBNoSQLSemanticCache client, connection string or endpoint must be set."
+      );
+    }
+
+    if (!dbConfig.client) {
+      if (connectionString) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        let [endpoint, key] = connectionString!.split(";");
+        [, endpoint] = endpoint.split("=");
+        [, key] = key.split("=");
+
+        client = new CosmosClient({
+          endpoint,
+          key,
+          userAgentSuffix: USER_AGENT_SUFFIX,
+        });
+      } else {
+        // Use managed identity
+        client = new CosmosClient({
+          endpoint,
+          aadCredentials: dbConfig.credentials ?? new DefaultAzureCredential(),
+          userAgentSuffix: USER_AGENT_SUFFIX,
+        } as CosmosClientOptions);
+      }
+    } else {
+      client = dbConfig.client;
+    }
+
+    this.config = {
+      ...dbConfig,
+      client,
+      databaseName: dbConfig.databaseName ?? DEFAULT_DATABASE_NAME,
```

Comment:
```suggestion
      databaseName: dbConfig.databaseName,
```
If param is not defined, will fall back to vector search default